### DISCUSSION TITLE
Relation field is not an foreign key?

### DISCUSSION THREAD

**Discussion by osseonews** [pocketbase/pocketbase#565] [Q&A]
I just opened our database to try to better understand how our tables were set up in Pocketbase, and I noticed that none of the relation fields we created in pocketbase are actually set up with foreign key constraints. Relation fields are all just text fields storing the id of the related collection, but there is no actual foreign key constraint, so in theory any value can go in the relation field. I assume this is intentional, and if so, what is the reasoning behind it? Isn't the power of relational databases based on these types of constraints, like a foreign key? How does Pocketbase ensure data integrity for relation fields if no foreign key constraint is present? Just relies on front end values? Thanks.

**Comment by ganigeorgiev** [comment]
The "foreigh key"-like constraint and validations for the relation fields are handled on application level because the relation is dynamic (eg. you are allowed to have a relation to single or multiple records, change the relation colleciton, or even change the cascade behavior).

**There is always server-side validation.** The frontend validation is just an addition for better UX.

**Reply by m-2k** [reply]
Interesting!

@ganigeorgiev Is it possible to add this information to the documentation, as well as how to use relationships between tables more efficiently and make subsequent selections (for example: do you need to create indexes in order to be able to quickly select from related tables; use views or APIs with record expansion)?

I was also surprised when I didn’t find foreign keys in the schema, after which I came across an article on the Internet that describes the reasons for the absence of foreign keys, and then came here and found a this question that I myself wanted to ask.
